test/document_stores/test_base.py

```diff
@@ -142,27 +142,41 @@ def test_get_all_documents_with_incorrect_filter_value(self, ds, documents):
         assert len(result) == 0
 
     @pytest.mark.integration
-    def test_extended_filter(self, ds, documents):
```

Comment:
This function was too big in any case, it was ported as it was in the first step of the refactoring but breaking it down was functional to this PR as the docstore doesn't fully support metadata and some tests are expected not to pass